packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -633,22 +673,31 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
 
   @override
   double? computeDistanceToActualBaseline(TextBaseline baseline) {
-    if (_direction == Axis.horizontal) {
-      return defaultComputeDistanceToHighestActualBaseline(baseline);
-    }
-    return defaultComputeDistanceToFirstActualBaseline(baseline);
+    // `defaultComputeDistanceToFirstActualBaseline` does not work even for
+    // Columns, because the children reorder with VerticalDirection.up.
```

Comment:
Was it a bug that we used defaultComputeDistanceToFirstActualBaseline here before?

Reply:
Yes. there's a test that would fail if this diff gets reverted.

Reply:
need to revert this change because of G3 failures. Will create an issue.

Reply:
https://github.com/flutter/flutter/issues/145739